src/plugins/intel_npu/src/al/include/intel_npu/al/icompiler.hpp

```diff
@@ -196,6 +200,12 @@ class ICompiler : public std::enable_shared_from_this<ICompiler> {
                                                                     const std::vector<uint8_t>& network,
                                                                     const Config& config) const = 0;
 
+    // Needed only by the driver compiler, to release the graph handle
+    virtual void release([[maybe_unused]] std::shared_ptr<const NetworkDescription> networkDescription){};
```

Comment:
We should not extend ICompiler, as this is implemented by the actual compiler and it should not be coupled with L0 flow at all.

Reply:
or it may be part of the executor itself. Double-check how this should be done in the case of the MLIR.

Reply:
We concluded we cannot do this in the executor because we have scenarios where the executor is not created at all ( eg. offline compilation with CREATE_EXECUTOR=0). We agreed we need to refactor this in another PR to avoid having this functions in the CIP interface.